crates/rspack_plugin_javascript/src/plugin/mod.rs

```diff
@@ -39,9 +52,60 @@ use crate::runtime::{
 static COMPILATION_DRIVES_MAP: Lazy<FxDashMap<CompilationId, JavascriptModulesPluginPluginDrive>> =
   Lazy::new(Default::default);
 
+#[derive(Debug, Clone)]
+struct WithCode<T> {
+  code: String,
+  value: T,
+}
+
+#[derive(Debug, Default)]
+struct RenameModuleCache {
+  inlined_modules_to_info: FxDashMap<String, WithCode<InlinedModuleInfo>>,
```

Comment:
Let's use `module.build_info.hash` as the key here instead of module identifier, so the cache can invalidate when the original source of the module changes

Reply:
`(module.identifier, module.build_info.hash)`

Reply:
If we using `module.build_info.hash` here, how to clear the outdated cache of a certain module which could be edited by user for many times.

Reply:
I see.. there is `code.source() == ident_info_with_code.code` to do the invalidate, LGTM

Reply:
Could we compare `build_info.hash` instead of source code? Seems like the overhead is cost less, the source code could be giant.

Reply:
Done, use hash instead of source code to compare.